### DISCUSSION TITLE
Always load primary key

### DISCUSSION THREAD

**Discussion by ronaldvdv** [ChilliCream/hotchocolate#3582] [Q&A] [answered]
I'm using Hot Chocolate with Entity Framework (Core). I would like to extend my entities using `ExtendObjectType` to define some additional fields, which depend on fields and relationships on the original entities.

In my particular scenario, I have a `Season` which contains multiple `Episode`. The `Season` has an `Index`; the `Episode` has a `Title` and an `Index` too. Both entities have a `Id`.

Now I would like to define two additional, calculated fields on the `Episode` which do not exist in the database:
- `Slug` based on the `Title`
- `Code` based on the `Index` and the `Season.Index`

I would use `ExtendObjectType(typeof(Episode))` on some class `EpisodeExtensions` and define the two new fields there as methods which have an argument `[Parent] Episode episode `.

However, this `episode` variable will only have the fields populated which are specified in the GraphQL query. So if the GraphQL query would _only_ contain `Slug` and `Code` and _not_ include `Title`, `Index` and `Season.Index`, there is no way to calculate `Slug` and `Code` in the resolver. In fact, it seems `Episode.Id` and `Episode.SeasonId` are not populated either, so it is impossible to access the full `Episode` and its `Season`.

What would be the right way to guarantee that more data from the original entity can be accessed, at least the `Episode.Id`?

_Note_ I believe the same problem would occur if I had defined the resolvers for the two additional fields on the entities themselves, but I want to make sure I describe the scenario as I'm using it and I've defined my entity classes in a separate project with no Hot Chocolate dependencies.

**Comment by PascalSenn** [comment, marked as answer]
@ronaldvdv  
You can do 
```
public class Episode 
{
   [IsProjected(true)]
   public Guid Id {get;set;}
}
```

This will always project the ID

**Reply by ronaldvdv** [reply]
Brilliant, thanks!
